File: cs/DressDiscover/DressDiscover.Portable/src/DressDiscover/Gui/Portable/Services/Worksheet/LocalWorksheetCommandService.cs
using DressDiscover.Api.Models.Worksheet;
using DressDiscover.Api.Services.Worksheet;
using System.Collections.Generic;
using System.Diagnostics;

namespace DressDiscover.Gui.Portable.Services.Worksheet
{
    public sealed class LocalWorksheetCommandService : IWorksheetCommandService
    {
        public LocalWorksheetCommandService(IDictionary<string, object> properties)
        {
            this.properties = properties;
        }

        public void DeleteWorksheetState(string accessionNumber)
        {
            Debug.WriteLine("delete " + accessionNumber);
            properties.Remove(LocalWorksheetQueryService.GetWorksheetStateItemKey(accessionNumber));
        }

        public void PutWorksheetState(WorksheetState state)
        {
            var stateJsonString = state.ToJson().ToString(Newtonsoft.Json.Formatting.None);
            Debug.WriteLine("put {0} <- {1}", state.AccessionNumber, stateJsonString);
            properties[LocalWorksheetQueryService.GetWorksheetStateItemKey(state.AccessionNumber)] = stateJsonString;
        }

        private IDictionary<string, object> properties;
    }
}

File: ts/worksheet/src/dressdiscover/gui/worksheet/services/local_worksheet_command_service.ts
﻿import { AsyncToSyncWorksheetCommandService } from "dressdiscover/api/services/worksheet/async_to_sync_worksheet_command_service";
import { LocalWorksheetQueryService } from "dressdiscover/gui/worksheet/services/local_worksheet_query_service";
import { WorksheetState } from "dressdiscover/api/models/worksheet/worksheet_state";

export class LocalWorksheetCommandService extends AsyncToSyncWorksheetCommandService {
    deleteWorksheetStateSync(kwds: { accessionNumber: string }): void {
        console.info("delete " + kwds.accessionNumber);
        localStorage.removeItem(LocalWorksheetQueryService.getWorksheetStateItemKey(kwds.accessionNumber));
    }

    putWorksheetStateSync(kwds: { state: WorksheetState }): void {
        let jsonString = JSON.stringify(kwds.state.toThryftJSON());
        console.info("put " + jsonString);
        localStorage.setItem(LocalWorksheetQueryService.getWorksheetStateItemKey(kwds.state.accessionNumber), jsonString);
    }
}